dagster-io/create-react-app
# Produce CSP output file for production, bump to 5.0.3

In order to use our configured CSP as a header value, emit an output file that can be read by the server. This allows us to share CSP configuration between dev (via meta tag, the webpack plugin default) and prod (via header).

This approach is based on https://github.com/slackhq/csp-html-webpack-plugin#generating-a-file-containing-the-csp-directives.

- If an `outputFilename` is specified, and the current env is prod, the file will be emitted with the built CSP directives.
- If no `outputFilename` is specified, this is not done, and the default `processFn` is used. This creates a meta tag.
- If no CSP configuration is provided at all, we'll skip the plugin entirely. (This is currently broken, and requires that all `.dagster.js` must return something valid from the `csp` function.)

### Test plan

Link dagit-app to local CRA.

Test status quo:

- Run `yarn build` with current `.dagster.js`, e.g. no `outputFilename`. Run dagit with this build. Verify that the `<meta>` CSP is correctly in place.
- With dagit backend running, run webpack dev server. Verify that the `<meta>` CSP is correctly in place.

Test with new behavior:

- Add `outputFilename` to `.dagster.js` and behavior in `webserver.py` to generate the header based on the file. Run `yarn build`. Run dagit, verify that the CSP is now present as a header.
- With dagit backend running, run webpack dev server. Verify that the `<meta>` CSP remains correctly in place.
